### PR TITLE
Feature/update moving model to latest design

### PR DESCRIPTION
**Pull request by brettminnie**

### CODE REVIEW

**Comment by eddiejaoude** [comment]
Build failed, otherwise looks ok.

```
1) TransformCore\PHE\HayApi\PersistenceBundle\Tests\Controller\AboutYouControllerTest::testPutDependantsActionWithValidIdReturns200
Argument 1 passed to Doctrine\Common\Collections\ArrayCollection::__construct() must be of the type array, object given, called in /home/travis/build/TransformCore/HayPersistenceApi/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php on line 605 and defined
```

**Comment by brettminnie** [comment]
Still in development, I really must update the tags

**Comment by eddiejaoude** [comment]
Yep, labels/tags help ;)

File: src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Helper/DayDuration.php [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+ * @ORM\Entity
+ * @ORM\ChangeTrackingPolicy("DEFERRED_EXPLICIT")
+ * Class DayDuration
+ * @package TransformCore\PHE\HayApi\PersistenceBundle\Entity\Questions
+ */
+class DayDuration implements IsIdentifiable
+{
+    use Identifier;
+
+    protected static $validDays = array ('mon', 'tues', 'wed', 'thurs', 'fri', 'sat', 'sun');
```
Comment (eddiejaoude) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Helper/DayDuration.php]: Worth having a type? `array`

File: src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Moving.php [hunk 2]
```diff
@@ -22,23 +24,36 @@
...
+     *     cascade={"persist"}
+     * )
+     * @ORM\JoinColumn(name="moving_duration_id", referencedColumnName="id")
+     * @ORM\OrderBy({"id"="ASC"})
+     * @var MovingDuration
+     *
+     * @Exclude
      */
-    protected $aerobicActivityDays = 0;
+    protected $aerobicDayDurations;
```
Comment (eddiejaoude) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Moving.php]: Why default got removed?

File: src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Moving.php [hunk 3]
```diff
@@ -56,61 +71,87 @@ class Moving extends Section
...
+                $total ++;
+            }
+        }
+
+        return $total;
     }
 
     /**
-     * @param int $aerobicActivityDays
+     * @param array $aerobicActivityDays
```
Comment (eddiejaoude) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Moving.php]: Return `array`? Below on line 124, its is `return $this`

File: src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Helper/DayDuration.php (hunk 1 above)
Comment (brettminnie) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Helper/DayDuration.php]: Good point, will update

File: src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Moving.php (hunk 3 above)
Comment (brettminnie) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Moving.php]: Good spot, tiredness kills :D

File: src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Moving.php (hunk 2 above)
Comment (brettminnie) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Moving.php]: Read the docbloc :P Was an integer, now is a many to many relationship, it is now initialised in the object constructor

File: src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Moving.php (hunk 3 above)
Comment (eddiejaoude) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Moving.php]: :smile:

File: src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Moving.php (hunk 2 above)
Comment (eddiejaoude) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Moving.php]: ok :+1:

**Comment by brettminnie** [comment]
Merge away then plox

**Pull request merged**